src/request/src/middleware/src/http.js

```diff
@@ -1,56 +1,70 @@
-import { TimeoutError } from '../../../../errors';
 import Middleware from './middleware';
 import Promise from 'es6-promise';
-import agent from 'superagent';
+import httpRequest from 'request';
+import { isDefined } from 'src/utils';
+import pkg from 'package.json';
+
+function deviceInformation() {
+  const platform = process.title;
+  const version = process.version;
+  const manufacturer = process.platform;
+
+  // Return the device information string.
+  const parts = [`js-${pkg.name}/${pkg.version}`];
+
+  return parts.concat([platform, version, manufacturer]).map((part) => {
+    if (part) {
+      return part.toString().replace(/\s/g, '_').toLowerCase();
+    }
+
+    return 'unknown';
+  }).join(' ');
+}
 
 export default class HttpMiddleware extends Middleware {
   constructor(name = 'Http Middleware') {
     super(name);
   }
 
+  get deviceInformation() {
+    return deviceInformation();
+  }
+
   handle(request) {
     const promise = new Promise((resolve, reject) => {
       const { url, method, headers, body, timeout, followRedirect } = request;
-      const redirects = followRedirect === true ? 5 : 0;
-
-      this.httpRequest = agent(method, url)
-        .set(headers)
-        .send(body)
-        .timeout(timeout)
-        .redirects(redirects)
-        .end((error, response) => {
-          this.httpRequest = undefined;
-
-          if (error) {
-            response = error.response;
-          }
-
-          if (!response) {
-            if (error) {
-              if (error.code === 'ECONNABORTED') {
-                return reject(new TimeoutError(undefined, undefined, error.code));
-              }
-            }
 
-            return reject(error);
-          }
+      // Add the X-Kinvey-Device-Information header
+      headers['X-Kinvey-Device-Information'] = this.deviceInformation;
 
-          return resolve({
+      httpRequest({
+        method: method,
+        url: url,
+        headers: headers,
+        body: body,
+        followRedirect: followRedirect,
+        timeout: timeout
+      }, (error, response, body) => {
+        if (isDefined(response) === false) {
```

Comment:
@tejasranade I added some additional unit tests to cover different kinds of requests.